File: app/src/main/java/com/reddittask/android/ui/activity/main/MainPresenter.java
package com.reddittask.android.ui.activity.main;

import com.reddittask.android.model.GameData;
import com.reddittask.android.ui.activity.base.MvpPresenter;
import com.reddittask.android.ui.activity.base.MvpView;

/**
 * Created by marinaracu on 30.11.2017.
 */

public interface MainPresenter<V extends MainView> extends MvpPresenter<V>{

    void onItemNewsClicked(int position, String url);

    void loadNextData(int dataCountLimit);

}

File: app/src/main/java/com/keddits/ui/activity/main/MainPresenter.kt
package com.keddits.ui.activity.main

import com.keddits.model.Child
import com.keddits.ui.activity.base.MvpPresenter

/**
 * Created by marinaracu on 25.12.2017.
 */
interface MainPresenter<V : MainView> : MvpPresenter<V> {

    fun onItemNewsClicked(url: String)

    fun loadNextData(dataCountLimit: Int, exitingItems: Int)

    fun loadLastData(initialList: MutableList<Child>, skipCount: Int)
}